gotham/src/handler/assets/mod.rs

```diff
@@ -206,16 +207,26 @@ fn create_file_response(options: FileOptions, state: State) -> Pin<Box<HandlerFu
 
     let (path, encoding) = check_compressed_options(&options, &headers);
 
-    let response_future = File::open(path).and_then(|file| async move {
+    let response_future = File::open(path).and_then(move |mut file| async move {
         let meta = file.metadata().await?;
         if not_modified(&meta, &headers) {
             return Ok(hyper::Response::builder()
                 .status(StatusCode::NOT_MODIFIED)
                 .body(Body::empty())
                 .unwrap());
         }
-        let len = meta.len();
         let buf_size = optimal_buf_size(&meta);
+        let resolve_range_result = resolve_range(meta.len(), &headers);
+        if let Err(e) = resolve_range_result {
+            return Ok(hyper::Response::builder()
+                .status(StatusCode::RANGE_NOT_SATISFIABLE)
+                .body(Body::from(e))
+                .unwrap());
+        }
+        let (len, range_start) = resolve_range_result.unwrap();
```

Comment:
```suggestion
        let (len, range_start) = match resolve_range(meta.len(), &headers) {
            Ok((len, range_start)) => (len, range_start),
            Err(e) => {
                return Ok(hyper::Response::builder()
                    .status(StatusCode::RANGE_NOT_SATISFIABLE)
                    .body(Body::from(e))
                    .unwrap());
            }
        };
```

No need to use `.unwrap()` if the compiler can prove it.